README.md

```diff
@@ -420,152 +811,209 @@ A candidate will be selected if:
 
-```
-cd /path/to/my/DeezyMatch
-pip install -r requirements.txt
+```python
+myranker.output
 ```
 
-6. Install DeezyMatch:
+All the query-related parameters can be changed via `set_query` method, for example:
 
+```python
+myranker.set_query(query=["another_example"])
 ```
-cd /path/to/my/DeezyMatch
-python setup.py install
+
+other parameters include: 
+```bash
+query
+query_scenario
+ranking_metric
+selection_threshold
+num_candidates
+search_size
+number_test_rows
+output_path
 ```
 
-Alternatively:
+Again, we can rank the candidates for the new query by:
 
-```
-cd /path/to/my/DeezyMatch
-pip install -v -e .
+```python
+myranker.rank()
+# to access output:
+myranker.output
 ```
 
-7. Continue with the [Tutorial](#run-deezymatch-as-a-python-module-or-via-command-line)!
+## Tips / Suggestions on DeezyMatch functionalities
 
----
+### Candidate ranker
 
-:warning: If you get `ModuleNotFoundError: No module named '_swigfaiss'` error when running `candidateRanker.py`, one way to solve this issue is by:
+* As already mentioned, based on our experiments, `conf` is not a good metric for ranking candidates. Consider using `faiss` or `cosine`.
 
-```bash
-pip install faiss-cpu --no-cache
-```
+* Adding prefix/suffix to input strings (see `prefix_suffix` option in the input file) can greatly enhance the ranking results. However, we recommend one-character-long prefix/suffix; otherwise, this may affect the computation time.
 
-Refer to [this page](https://github.com/facebookresearch/faiss/issues/821).
+* In `candidate_ranker`, the user specifies a `ranking_metric` based on which the candidates are selected and ranked. However, DeezyMatch also reports the values of other metrics for those candidates. For example, if the user selects `ranking_metric="faiss"`, the candidates are selected based on the `faiss`-distance metric. At the same time, the values of `cosine` and `conf` metrics for **those candidates** (ranked according to the selected metric, in this case faiss) are also reported.
+
+* What is the role of `search_size` in candidate ranker?
+
+For a given query, DeezyMatch searches for candidates iteratively. At each iteration (i.e., one colored region in the figure below), the query vector is compared with five potential candidate vectors according to the selected ranking metric (`ranking_metric` argument). Here, `search_size` is five. In step-1, the five closest candidate vectors, as measured by L2-norm distance, are examined. Four (out of five) candidate vectors passed the threshold (specified by `selection_threshold` argument) in the figure (step-1). However, in this example, we assume `num_candidates` is 10. So, DeezyMatch examines the second batch of potential candidates, again five vectors (as specified by `search_size`). Three (out of five) candidates pass the threshold in step-2. Finally, in the third iteration, three more candidates are found. DeezyMatch collects the information of these ten candidates and go to the next query.
```

Comment:
The initial sentence where you say " is compared with five potential candidate vectors" I found it a bit confusing, as it seems from there that the number 5 is hardcoded. Maybe if you just mention before that sentence the piece where you say " Here, `search_size` is five" that would make it immediately clearer.

For instance:
"If we set `search_size` to **five**, at each iteration (i.e., one colored region in the figure below), the query vector is compared with **five** potential candidate vectors according to the selected ranking metric (`ranking_metric` argument)."

Reply:
Done: https://github.com/Living-with-machines/DeezyMatch/pull/68/commits/d96bbcaeb48e5204d2b2878df01c96205c894bb8